src/v/cloud_storage/remote.cc

```diff
@@ -107,10 +107,9 @@ static error_outcome categorize_error(
         // - any filesystem error
         // - broken-pipe
         // - any other network error (no memory, bad socket, etc)
-        if (auto code = cerr.code();
-            code.value() != ECONNREFUSED && code.value() != ENETUNREACH
-            && code.value() != ETIMEDOUT && code.value() != ECONNRESET
-            && code.value() != EPIPE) {
+        if (auto code = abs(cerr.code().value());
+            code != ECONNREFUSED && code != ENETUNREACH && code != ETIMEDOUT
+            && code != ECONNRESET && code != EPIPE && code != EBADR) {
```

Comment:
Hmm, I'm confused by the commit message. how is this error:

> error GnuTLS:-110, The TLS connection was non-properly terminated.

related to handling for standard errno.h codes like ETIMEDOUT? i was under the impression that gnutls error codes were unrelated to standard error codes.

Reply:
should this be a function instead? (so expanding it is just adding one element to the array + we can add a unit test. 

``` 
static_array<int> codes {{ refused, reset, unreach, timedout, badaddr, pipe }}

for (c in code) {
if code == c; return true
}
return false
``` 

kinda thing

Reply:
also, is nice if it's a func in case at some point we do openssl

Reply:
Oh, you're right.  Good catch.

I was assuming that something was munging gnutls errors into vaguely-semantically-equivalent posix errnos, but in fact they're totally separate.

I guess I need to revise this to do a disgusting string comparison on the `system_error`'s message to identify gnutls cases, and then compare them with official gnutls error codes from their header.

Reply:
You're right. It's not. I assumed gnutls maps it's error codes to linux error codes, but uses the negative for some reason. That doesn't seem to be the case. I guess we should silence specific gnutls errors.

Reply:
@jcsp perhaps you don't need to parse the string. if these gnutls-errors-as-std::system_error are originating from seastar, then I see that seastar is already attaching a `static const gnutls_error_category glts_errorc;` so we can identify which are errno.h and which are gnutls.